api-sandbox/Web3.elm

```diff
@@ -1,16 +1,26 @@
-module Web3 exposing (..)
+port module Web3
+    exposing
+        ( Web3State
+        , Request
+        , Response
+        , init
+        , handleResponse
+        , getBlockNumber
+        , request
+        , response
+        )
 
 import Dict exposing (..)
 import Json.Decode exposing (..)
 
 
-type alias Model =
-    { requestMap : Dict Int (Decoder Web3Response) }
+init : Web3State msg
+init =
+    Web3State Dict.empty
 
 
-init : Model
-init =
-    { requestMap = Dict.empty }
+type Web3State msg
+    = Web3State (Dict Int (String -> msg))
```

Comment:
So this is turning out to be a bit more difficult than I thought originally. The idea is that we want a way to lookup the `Msg` type value that will be threaded back through our `update` function, but if we just leave this as `Dict Int msg`, the compiler doesn't have a way to type-check all possible web3 result types we might want to decode and handle, so I had to fix it to just string responses right now (and that's why `result` on `Response` is a String)